modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/ScaledFloatFieldMapper.java

```diff
@@ -289,8 +289,9 @@ public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower
                 @Override
                 public IndexFieldData<?> build(IndexSettings indexSettings, MappedFieldType fieldType, IndexFieldDataCache cache,
                         CircuitBreakerService breakerService, MapperService mapperService) {
-                    final IndexNumericFieldData scaledValues = (IndexNumericFieldData) new DocValuesIndexFieldData.Builder()
-                            .numericType(IndexNumericFieldData.NumericType.LONG)
+                    final IndexNumericFieldData scaledValues = (IndexNumericFieldData) new SortedNumericIndexFieldData.Builder(
```

Comment:
Could the builder return a `IndexNumericFieldData`?

Reply:
Like, not all the builders can, but I'm fairly sure the `SortedNumericIndexFieldData` builder will only ever build numeric field data.